src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/resource/ExampleSecondResource.kt

```diff
@@ -0,0 +1,26 @@
+package uk.gov.justice.digital.hmpps.hmppstemplatepackagename.resource
+
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.hmppstemplatepackagename.service.UserMessageDto
+import uk.gov.justice.hmpps.kotlin.auth.AuthSource
+
+@RestController
+// Protected by a read only role
+@PreAuthorize("hasRole('ROLE_TEMPLATE_KOTLIN__RO')")
+@RequestMapping(value = ["/example-2"], produces = ["application/json"])
+class ExampleSecondResource {
+
+  @GetMapping("/{authSource}")
+  fun getSecondExampleUserMessage(@PathVariable authSource: AuthSource): UserMessageDto = when (authSource) {
+    AuthSource.NOMIS -> "Found a DPS user"
+    AuthSource.DELIUS -> "Found a Delius user"
+    AuthSource.AUTH -> "Found an external user"
+    else -> "Found a user that authenticated elsewhere"
+  }.let {
+    UserMessageDto(it)
+  }
+}
```

Comment:
I wonder if we could name the examples better. Maybe `ExampleUiResource` and `ExampleApiResource`? Although technically there is no such difference it helps to illustrate what the examples are for?

Though I'm not very convinced by my own argument. Maybe a comment in each file that explains what each resource is for? 

The confusing thing is that the UI resource endpoint calls back to the API resource endpoint - I can see why it needs to do that, but wonder if it will confuse people.

Reply:
👍 for adding a comment.  I'm slightly worried that calling it `xxxUiResource` will suggest to people that they need to add in `UiResource` if they're call it from the front end and `ApiResource` if it is only called from other APIs.  I wanted to call it `ExampleResource` and `ExampleActuallyThisShouldBeInADifferentServiceFromTheKotlinTemplateResource` but thought that it was too long...

Reply:
`ExampleActuallyThisShouldBeInADifferentServiceFromTheKotlinTemplateResource` does exactly what it says on the tin! 😄 More comments sounds sensible.